app/routes.py

```diff
@@ -1,2 +1,333 @@
+    
+    if "title" not in request_body or "description" not in request_body or "completed_at" not in request_body:
+        return jsonify({"details": "Invalid data"}), 400
+    
+    new_task = Task(
+        title=request_body["title"], 
+        description=request_body["description"], 
+        completed_at=request_body["completed_at"]
+        )
+    
+    db.session.add(new_task) 
+    db.session.commit()
+    
+    response_body = {
+        "task": {
+            "id": new_task.task_id,
+            "title": new_task.title,
+            "description": new_task.description,
+            "is_complete": new_task.completed_at is not None
+            }
+        }
+    
+    return jsonify(response_body), 201
+
+@tasks_bp.route("", methods=["GET"])
+def read_tasks():
+    title_query = request.args.get("sort")
+
+    if title_query == "asc":
+        tasks = Task.query.order_by(Task.title.asc())
+    elif title_query == "desc":
+        tasks = Task.query.order_by(Task.title.desc())
+    else:
+        tasks = Task.query.all() 
+    
+    task_responses = []
+    for task in tasks:
+        task_responses.append(
+                {
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": task.completed_at is not None
+                }
+            )
+    
+    return jsonify(task_responses), 200
+
+@tasks_bp.route("/<task_id>", methods=["GET"])
+def read_task(task_id):
+    task_id = int(task_id)
+    task = Task.query.get(task_id)
+    
+    if task == None:
+        return jsonify(None), 404
+
+    if not task.goal_id:
+        return jsonify({
+            "task": {
+            "id": task.task_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": False if task.completed_at == None else True
+        }
+        }), 200
+    else:
+        return jsonify({
+            "task": {
+            "id": task.task_id,
+            "goal_id": task.goal_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": False if task.completed_at == None else True
+        }
+        }), 200
+
+@tasks_bp.route("/<task_id>",methods=["PUT"])
+def update_task(task_id):
+    task_id = int(task_id)
+    task = Task.query.get(task_id)
+
+    updated_body = request.get_json()
+
+    if task == None:
+        return jsonify(None), 404
+
+    else: 
+        task.title = updated_body["title"]
+        task.description = updated_body["description"]
+        
+        db.session.commit()
+        
+        task_dict = {}
+        task_dict = {
+            "task": {
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": task.completed_at is not None
+            }
+        }
+        
+        return jsonify(task_dict), 200
```

Comment:
This logic doesn't need to be wrapped in an else clause. You can omit the else clause, unindent this code, and see that it works the same way.